src/librustc_typeck/check/callee.rs

```diff
@@ -21,11 +21,29 @@ use rustc_target::spec::abi;
 /// Checks that it is legal to call methods of the trait corresponding
 /// to `trait_id` (this only cares about the trait, not the specific
 /// method that is called).
-pub fn check_legal_trait_for_method_call(tcx: TyCtxt<'_>, span: Span, trait_id: DefId) {
+pub fn check_legal_trait_for_method_call(
+    tcx: TyCtxt<'_>,
+    span: Span,
+    receiver: Option<Span>,
+    trait_id: DefId,
+) {
     if tcx.lang_items().drop_trait() == Some(trait_id) {
-        struct_span_err!(tcx.sess, span, E0040, "explicit use of destructor method")
-            .span_label(span, "explicit destructor calls not allowed")
-            .emit();
+        let mut err = struct_span_err!(tcx.sess, span, E0040, "explicit use of destructor method");
+        err.span_label(span, "explicit destructor calls not allowed");
+
+        let snippet = receiver
+            .and_then(|s| tcx.sess.source_map().span_to_snippet(s).ok())
+            .unwrap_or_default();
+
+        let (suggestion, applicability) = if snippet.is_empty() {
+            (snippet, Applicability::Unspecified)
+        } else {
+            (format!("drop({})", snippet), Applicability::MachineApplicable)
+        };
+
+        err.span_suggestion(span, "consider using `drop` function", suggestion, applicability);
```

Comment:
If you're suggesting `drop(x)` then the span for the suggestion will need to be the one for the whole method call.

Reply:
I have tried to merge spans, but I'm suppose it still won't do because the resulting span doesn't include round brackets: `x.drop` instead of `x.drop()`. Is there an easy way to achieve that?

Reply:
Perhaps it isn't worth the fuss and because applicability still would be `MaybeIncorrect` at best, I can simply print a message leaving the suggestion empty. Something like that:
```
let message = format!("consider using `drop` function{}", snippet);

err.span_suggestion(
    span,
    &message,
    String::new(),
    Applicability::Unspecified,
);
```
What do you think?

Reply:
@matthewjasper Sorry for bothering you, but I'm not sure if I should try to extend the span somehow to include the whole method call or use a simpler solution?

Reply:
I have updated the pull request accordingly.